lib/resilient/circuit_breaker.rb

```diff
@@ -1,9 +1,33 @@
 require "resilient/key"
 require "resilient/circuit_breaker/metrics"
 require "resilient/circuit_breaker/properties"
+require "resilient/circuit_breaker/registry"
 
 module Resilient
   class CircuitBreaker
+    # Public: Resets all registered circuit breakers (and thus there state/metrics).
+    # Useful for ensuring a clean environment between tests.
+    def self.reset
+      registry.reset
+    end
+
+    # Private: Registry that stores instances of circuit breakers to ensure that
+    # attempts to create new instances with the same key retain existing state/metrics.
+    def self.registry
+      @registry ||= Registry.new
+    end
+    class << self; private :registry; end
+
+    # Public: Overriden method to return instance of circuit breaker based on
+    # key. If key does not exist, it is registered. If key does exist, it
+    # returns registered instance instead of allocating a new instance in order
+    # to ensure that state/metrics are the same per key.
+    def self.new(key: nil, open: false, properties: nil, metrics: nil)
```

Comment:
I go back and forth on this every time too. Right now I'm on the "`new` should create new instances" side but not real strongly. One alternative is to create the new constructor, like `Resilient::CircuitBreaker.for(...)` and mark `new` as private.

I sometimes regret a static registry. I've done something like this in the past:

```ruby
class Registry
  @default = new

  def self.default
    @default
  end

  # ...
end

class Breaker
  def self.new
    raise "new breakers can't be created directly"
  end

  def self.for(key:, registry: nil)
    (registry || Registry.default).fetch(key) { new(key) }
  end
end
```